R/reshape.R

```diff
@@ -116,16 +119,23 @@ focus_if.default <- function(x, .predicate, ..., mirror = FALSE) {
 #' x <- shave(x)  # use shave to set upper triangle to NA and then...
 #' stretch(x, na.rm = FALSE)  # omit all NAs, therefore keeping each
 #'                              # correlation only once.
-stretch <- function(x, na.rm = FALSE, remove.dups =  FALSE) {
+stretch <- function(x, na.rm = FALSE, remove.dups =  FALSE,
+                    .order = c("default", "alphabet")) {
+  .order <- match.arg(.order)
   UseMethod("stretch")
 }
 
 #' @export
-stretch.cor_df <- function(x, na.rm = FALSE, remove.dups =  FALSE) {
+stretch.cor_df <- function(x, na.rm = FALSE, remove.dups =  FALSE,
+                           .order = c("default", "alphabet")) {
+  .order <- match.arg(.order)
   if(remove.dups) x <- shave(x)
   row_name <- x$rowname
   x <- x[, colnames(x) != "rowname"]
   tb <- imap_dfr(x, ~tibble(x = .y, y = row_name, r = .x))
+  if(.order == "default")  {
```

Comment:
These changes do not seem necessary to address the original issue. Rather, they would represent an expansion of the capabilities of `stretch`. If `stretch` is to be augmented to optionally sort in alphabetical order, another option may be to allow an alphabetical option to  `rearrange`.